docs/src/converting_zygoterules.md

```diff
@@ -0,0 +1,136 @@
+e.g. what is the Tangent to a String or an index: those can't be perturbed.
+
+See [FAQ on difference between `ZeroTangent` and `NoTangent`](@ref faq_abstract_zero).
+At the end of the day it doesn't matter too much if you get them wrong.
+`NoTangent` and `ZeroTangent` more or less act identically.
+
+### `Tuple`s and `NamedTuple`s become `Tangent{T}`s
+Zygote uses `Tuple`s and `NamedTuple`s to represent the structural tangents for `Tuple`s and `struct`s respectively.
+ChainRules core provides a generic `Tangent{T}`(@ref Tangent) to represent the structural tangent of a primal type `T`.
+It takes positional arguments if representing tangent for a `Tuple`.
+Or keyword argument to represent the tangent for a `struct`.
+When representing a `struct` you only need to list the nonzero fields -- any not given are implicit considered to be [`ZeroTangent`](@ref).
+
+When we say structural tangent we mean tangent types that are based only on the structure of the primal.
+This is in contrast to a natural tangent which captures some knowledge based on what the primal type represents.
+(E.g. for arrays a natural tangent is often the same kind of array).
+For more details see the the [design docs on the many tangent types](@ref manytypes)
+
+
+## Calling back into AD (`ZygoteRules.pullback`)
+Rules that need to call back into the AD system, e.g, for higher order functions like `map(f, xs)`, need to be changed.
+In `ZygoteRules` you can use `ZygoteRules.pullback` or `ZygoteRules._pullback`, which will always result in calling into Zygote.
+Since ChainRules is AD agnostic, you can't do that.
+Instead you use a [`RuleConfig`](@ref) to specify requirements of an AD system e.g `::RuleConfig{>:CanReverseMode}` work for Zygote,
+and then use [`rrule_via_ad`](@ref).
+
+See the [docs on calling back into AD](@ref config) for more details.
+
+## Consider adding some thunks
+
+A feature ChainRulesCore offers that ZygoteRules doesn't is support for thunks.
+Where work is delayed until it is needed, and avoided if it never is.
+See docs on [`@thunk`](@ref), [`Thunk`](@ref), [`InplaceableThunk`](@ref).
+
+You don't have to though.
+It is easy to go overboard with using thunks.
+
+## Testing Changes
+
+One of the advantages of using ChainRules is that you can easily and robustly test it with [ChainRulesTestUtils.jl](https://juliadiff.org/ChainRulesTestUtils.jl/stable/).
+This both uses finite differencing to test accuracy of derivative, as well as checking the correctness of the API.
+It should catch anything you might have gotten wrong referred to in this page.
+
+The test for the above example is `test_rrule(f, 2.5, Foo(9.9, 7.2), 31.0)`.
+You can see it looks a lot like an example call to `rrule`, just with the prefix `test_` added to the start.
+
+## `@nograd` becomes `@non_differentiable`
+Probably more or less with no changes.
+[`@non_differentiable`](@ref) also lets you specify a signature.
```

Comment:
```suggestion
[`@non_differentiable`](@ref) also lets you specify a signature in case you want to restrict non-differentiability to a certain subset of argument types.
```